src/test/java/uk/gov/ons/census/casesvc/messaging/SampleReceiverTest.java

```diff
@@ -87,11 +93,23 @@ public void testHappyPath() {
 
     // Then
     // Check the emitted event
-    ArgumentCaptor<CaseCreatedEvent> emittedMessageArgCaptor =
-        ArgumentCaptor.forClass(CaseCreatedEvent.class);
-    verify(rabbitTemplate)
+    ArgumentCaptor<FanoutEvent> emittedMessageArgCaptor =
+        ArgumentCaptor.forClass(FanoutEvent.class);
+    verify(rabbitTemplate, times(2))
         .convertAndSend(eq("myExchange"), eq(""), emittedMessageArgCaptor.capture());
-    CaseCreatedEvent caseCreatedEvent = emittedMessageArgCaptor.getValue();
+    List<FanoutEvent> fanoutEvents = emittedMessageArgCaptor.getAllValues();
```

Comment:
This test is now (and was before additions) very long and difficult to follow/alter.  And it's only testing the happy path.  Does it add a lot of value considering that the IT tests presumably cover the same ground with less effort/mocking/spying etc.

Is there anyway to refactor the code/test to make it 'nicer'

Reply:
check out my refactoring